bool control
float32 setpoint
